boxsdk/object/metadata_template.py

```diff
@@ -0,0 +1,282 @@
+            `unicode`
+        :param option_keys:
+            The option keys in the desired order
+        :type option_keys:
+            `list` of `unicode`
+        """
+        self.ops.append({
+            'op': 'reorderEnumOptions',
+            'fieldKey': field_key,
+            'enumOptionKeys': option_keys,
+        })
+
+    def reorder_fields(self, field_keys):
+        """
+        Reorders the fields in a metadata template, which affects their display in UI.
+
+        :param field_keys:
+            The field keys in the desired order
+        :type field_keys:
+            `list` of `unicode`
+        """
+        self.ops.append({
+            'op': 'reorderFields',
+            'fieldKeys': field_keys,
+        })
+
+    def edit_field(self, field_key, field):
+        """
+        Edits a field in the template.
+
+        :param field_key:
+            The key of the field to update
+        :type field_key:
+            `unicode`
+        :param field:
+            The updated field values
+        :type field:
+            :class:`MetadataField`
+        """
+        self.ops.append({
+            'op': 'editField',
+            'fieldKey': field_key,
+            'data': field.json(),
+        })
+
+    def edit_enum_option_key(self, field_key, old_option_key, new_option_key):
+        """
+        Change the key of an enum field option.
+
+        :param field_key:
+            The key of the template field in which the option appears
+        :type field_key:
+            `unicode`
+        :param old_option_key:
+            The old option key
+        :type old_option_key:
+            `unicode`
+        :param new_option_key:
+            The new option key
+        :type new_option_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'editEnumOption',
+            'fieldKey': field_key,
+            'enumOptionKey': old_option_key,
+            'data': {
+                'key': new_option_key,
+            },
+        })
+
+    def remove_enum_option(self, field_key, option_key):
+        """
+        Remove an option from an enum field.
+
+        :param field_key:
+            The key of the template field in which the option appears
+        :type field_key:
+            `unicode`
+        :param option_key:
+            The key of the enum option to remove
+        :type option_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'removeEnumOption',
+            'fieldKey': field_key,
+            'enumOptionKey': option_key,
+        })
+
+    def remove_field(self, field_key):
+        """
+        Remove a field from the metadata template.
+
+        :param field_key:
+            The key of the field to remove
+        :type field_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'removeField',
+            'fieldKey': field_key,
+        })
+
+
+class MetadataFieldType(TextEnum):
```

Comment:
is this only used in tests?

maybe it could be used in the docstring of the `MetadataField` constructor